iothub/device/src/TransportSettings/IotHubClientTransportSettings.cs

```diff
@@ -72,5 +73,63 @@ public override string ToString()
         {
             return $"{GetType().Name}/{Protocol}";
         }
+
+        /// <summary>
+        /// The acceptable versions of TLS to use when the SDK must be explicit.
+        /// </summary>
+        public SslProtocols MinimumTlsVersions { get; private set; } = SslProtocols.Tls12;
+
+        /// <summary>
+        /// The version of TLS to use by default.
+        /// </summary>
+        /// <remarks>
+        /// Defaults to "None", which means let the OS decide the proper TLS version (SChannel in Windows / OpenSSL in Linux).
+        /// </remarks>
+        public SslProtocols PreferredTlsVersions { get; private set; } = SslProtocols.None;
+
+#pragma warning disable CA5397 // Do not use deprecated SslProtocols values
+        private const SslProtocols AllowedProtocols = SslProtocols.Tls | SslProtocols.Tls11 | SslProtocols.Tls12;
+        private const SslProtocols PreferredProtocol = SslProtocols.Tls12;
+
+        /// <summary>
+        /// To enable certificate revocation check. Default to be false.
+        /// </summary>
+        public bool CertificateRevocationCheck { get; set; }
+
+        /// <summary>
+        /// Sets the minimum acceptable versions of TLS.
+        /// </summary>
+        /// <remarks>
+        /// Will ignore a version less than TLS 1.0.
+        ///
+        /// Affects:
+        /// 1. MinimumTlsVersions property
+        /// 2. Preferred property
+        /// 3. For .NET framework 4.5.1 over HTTPS or websocket, as it does not offer a "SystemDefault" option, the version must be explicit.
+        /// </remarks>
+        public void SetMinimumTlsVersions(SslProtocols protocols = SslProtocols.None)
```

Comment:
We still need this as it ignore versions less than TLS 1.0 and ensure that users don't use deprecated SslProtocols values.

Reply:
What do we care if they set a value they want? The service will reject versions it doesn't accept.